src/Service/Resolvers/SqlMutationEngine.cs

```diff
@@ -331,13 +331,22 @@ public class SqlMutationEngine : IMutationEngine
 
                 if (context.OperationType is Config.Operation.Insert)
                 {
-                    if (mutationResultRow is null || mutationResultRow.Columns.Count == 0)
+                    if (mutationResultRow is null)
```

Comment:
Can use this as opportunity to throw a dab exception. And better document why this case would ever happen.

Reply:
Added comment, but not throwing a DAB exception here as it is already being done in RestController class.

Reply:
While I concur that RestController is throwing the exception, it is neither adding nor communicating any messaging beyond "unexpected error" and a callstack. It would be helpful to throw a dab exception which contains a helpful/actionable error message.

Reply:
As I skimmed through stackoverflow and the codebase, I don't actually see this case coming into picture at all but most likely it will not be an actionable thing on the user side. Will need to investigate more. Basically this case will arise only when we hit this block:
https://github.com/Azure/data-api-builder/blob/7953d06c52eb0002c429a8c87b39bccf6ecc4225/src/Service/Resolvers/QueryExecutor.cs#L180

And this is hit when `dataReaderHandler is not null && dbDataReader is not null` . dbDataReaderHandler as we pass it, we know for sure it won't be null and for dbDataReader to be null, I couldn't find anything in the docs or stackoverflow.

Reply:
it might be worth exploring what happens when removing some of the nullable `?` operators from method signatures around where you were looking, and from any local variables to see what additional hints intellisense might provide

Reply:
discussed over call to have a better exception message.